view/src/components/DiffViewWrapper.jsx

```diff
@@ -87,39 +135,73 @@ function getSyntaxHighlightedTokens(hunks) {
+ * in that case set the code of the other version (oldCode/newCode) to empty string.
+ * The entire code of the function is shown then.
  *
  * For the lines which are not shown, we render buttons that allow user to show these lines.
- * @param {Object} props
+ * @param {Object} props - The old/new props can be null when showing only one version
+ *   of the function.
  * @param {string} props.oldCode - Source code of file containing old version of the function.
+ * @param {string} props.newCode - Source code of file containing new version of the function.
+ *   If oldCode or newCode is empty string, then we are showing only one version of function.
  * @param {string} props.diff - Output of unified diff of old and new source code.
  * @param {number} props.oldStart - Line where the old version of the function starts.
  * @param {number} props.newStart - Line where the new version of the function starts.
  * @param {number} props.oldEnd - Line where the old version of the function ends.
+ * @param {number} props.newEnd - Line where the new version of the function ends.
  * @param {Boolean} [props.showDiff=true] - True to color differences in function.
  * @param {Number} [props.linesToShow=null] - Lines (old, new) of function which
  *                                            should be shown and highlighted.
  */
 export default function DiffViewWrapper({
   oldCode,
+  newCode,
   diff,
   oldStart,
   newStart,
   oldEnd,
+  newEnd,
   showDiff = true,
   linesToShow = null,
 }) {
-  /* getting object which Diff component needs */
+  /* Showing code only for one version of a function? */
+  const onlyOneSide = (oldCode === '' || newCode === '');
+  /**
+   * Getting object which Diff component needs, it is an object which represents
+   * differences in a file, it contains hunks (group of changes),
+   * the hunks contain individual line changes (insert/delete/normal).
+   */
   const file = useMemo(() => {
-    if (diff !== '') {
+    let returnValue;
+    if (onlyOneSide) {
+      // Only one side/version of code will be shown - creating the object from the code.
+      const onlyOld = (newCode === '');
+      const [code, start, end] = onlyOld
+        ? [oldCode, oldStart, oldEnd]
+        : [newCode, newStart, newEnd];
+      returnValue = createFileStructForOneSide(onlyOld, code, start, end);
+    } else if (diff !== '') {
+      // Code contains syntax differences - creating the object from the diff.
       const [fileObject] = parseDiff(diff);
-      return fileObject;
+      returnValue = fileObject;
+    } else {
+      // Code does not contain syntax differences - creating the object using first line of code.
+      returnValue = createFileStructure(oldCode, oldStart, newStart);
     }
-    return createFileStructure(oldCode, oldStart, newStart);
-  }, [diff, oldCode, oldStart, newStart]);
-    /* hunks of differences/code which will be shown */
+    return returnValue;
```

Comment:
Perhaps it is because I use different style, but wouldn't multiple return statements be more readable then the returnValue? It is a small anonymous function, so your intent is clear.

Reply:
Thanks for the feedback. I originally wanted to write it as you mentioned, but the linter that we are using marked it as [no-else-return](https://eslint.org/docs/latest/rules/no-else-return) error, so this was my approach to overcome it. I changed it and now I am using `if` instead of `else if`, the linter is OK with this and hopefully it will be a little bit more readable.